### ISSUE TITLE
MongoDB test for smshandler

### ISSUE THREAD

**Issue by p4u** [vocdoni/blind-csp#16]
The smshandler can be run with two different storage layers (abstracted by the [Storage interface](https://github.com/vocdoni/blind-csp/blob/f/smshandler/handlers/smshandler/storage.go)). There are currently two implementations, a local KV store and a remote MongoDB.

There is a test that testes the storage implementation using the interface https://github.com/vocdoni/blind-csp/blob/f/smshandler/handlers/smshandler/storage_test.go

However we don't have currently any way to test the mongoDB storage implementation https://github.com/vocdoni/blind-csp/blob/f/smshandler/handlers/smshandler/mongodbstorage.go

We have several options (not sure which one is better):
1. Do a integration test based on docker-compose, run a MongoDB server and implement a testing main application (as we do in vocdoni-node)
2. Use Mocking
3. Use an in-memory mongoDB implementation for the unit test such as https://github.com/strikesecurity/strikememongo

@mvdan what do you think?

**Comment by altergui** [comment]
i'll give a stab at strikememongo first, which sounds leaner, if it doesn't make sense i'll fall back to docker-compose

**Comment by p4u** [comment]
After merging your PR I'm not sure this was a good solution.

```
memongo] [INFO]  Starting MongoDB with options &strikememongo.Options{ShouldUseReplica:false, Port:39201, CachePath:"/home/p4u/.cache/memongo", MongoVersion:"4.0.28", DownloadURL:"https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-4.0.28.tgz", MongodBin:"", Logger:(*log.Logger)(nil), LogLevel:0, StartupTimeout:10000000000}
[memongo] [INFO]  mongod from https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-4.0.28.tgz does not exist in cache, downloading to /home/p4u/.cache/memongo/mongodb-linux-x86_64-4_0_28_tgz_803c867fb4/mongod
--- FAIL: TestStorageMongoDB (116.01s)
    storage_test.go:25: error writing mongod binary from /tmp/220520311 to /home/p4u/.cache/memongo/mongodb-linux-x86_64-4_0_28_tgz_803c867fb4/mongod: rename /tmp/220520311 /home/p4u/.cache/memongo/mongodb-linux-x86_64-4_0_28_tgz_803c867fb4/mongod: invalid cross-device link
```
Looks like each time the test is run it downloads a mongo binary. That's too painful and we are making a unit test depend on Internet connection. @altergui @mvdan thoughts?

**Comment by altergui** [comment]
i proposed a fix upstream  https://github.com/strikesecurity/strikememongo/pull/6

but i'll anyway explore the docker alternative, since it shouldn't be too difficult

**Comment by altergui** [comment]
here's a replace line in the meantime
`replace "github.com/strikesecurity/strikememongo" v0.2.4 => "github.com/altergui/strikememongo" v0.2.4`

**Comment by p4u** [comment]
Can you check the branch `f/sms_queue` ?

**Comment by altergui** [comment]
> Same issue @altergui

that's bizarre. but anyway i just pushed an alternative, the challenge was simple and i wanted to play with managing docker containers from go, to see if it would make sense to rewrite vocdoni-node start_test.sh this way, and not deal with bash anymore.

**Comment by altergui** [comment]
#22 is closed but included in #24 , so i'm closing this issue
